src/graphics/VertexBuffer.h

```diff
@@ -56,19 +56,30 @@ struct VertexBufferDesc {
 	BufferUsage usage;
 };
 
-class Mappable {
+class Mappable : public RefCounted {
 public:
 	virtual ~Mappable() { }
 	virtual void Unmap() = 0;
 
+	inline Uint32 GetSize() const { return m_size; }
+	inline Uint32 GetCapacity() const { return m_capacity; }
+
 protected:
-	Mappable() : m_mapMode(BUFFER_MAP_NONE) { }
+	Mappable(const Uint32 size) : m_mapMode(BUFFER_MAP_NONE), m_size(size), m_capacity(size) { }
 	BufferMapMode m_mapMode; //tracking map state
+
+	// size is the current number of elements in the buffer
+	Uint32 m_size;
+	// capacity is the maximum number of elements that can be put in the buffer
+	Uint32 m_capacity;
+
+private:
+	Mappable() {}
```

Comment:
Nit: Why is this private constructor is needed? It looks like it can just be deleted.

Reply:
Probably can be deleted, at the time I was trying something out.